src/hotspot/share/cds/filemap.cpp

```diff
@@ -169,21 +170,44 @@ template <int N> static void get_header_version(char (&header_version) [N]) {
 FileMapInfo::FileMapInfo(bool is_static) {
   memset((void*)this, 0, sizeof(FileMapInfo));
   _is_static = is_static;
+  size_t c_header_size;
```

Comment:
FileMapInfo::FileMapInfo() is called in two places:

(1) When reading an archive -- _header should not be initialized here. It should be done inside FileMapInfo::init_from_file.

(2) When writing an archive -- I think we should move the _header initialization to FileMapInfo::populate_header().